packages/circuit-ui/components/SideNavigation/types.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright 2021, SumUp Ltd.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { HTMLProps, FC, SVGProps, MouseEvent, KeyboardEvent } from 'react';
+import { Dispatch as TrackingProps } from '@sumup/collector';
+
+export interface PrimaryLinkProps extends HTMLProps<HTMLAnchorElement> {
+  /**
+   * Display an icon in addition to the text to help to identify the link.
+   * On narrow viewports, only the icon is displayed.
+   */
+  icon: FC<SVGProps<SVGSVGElement>>;
+  /**
+   * Short label to describe the target of the link.
+   */
+  label: string;
+  /**
+   * Function that's called when the link is clicked.
+   */
+  onClick?: (event: MouseEvent | KeyboardEvent) => void;
+  /**
+   * Whether the link is the currently active page.
+   */
+  isActive?: boolean;
+  /**
+   * Whether the link is the currently active page.
+   */
+  isExternal?: boolean;
+  /**
+   * TODO: Add description
+   */
+  badge?: {
```

Comment:
Why did we make this one an object?

Reply:
In case we need to support further properties in the future, such as a `variant`.

Reply:
Oh please no

Reply:
But this makes sense 😄 avoids a breaking change